src/core/rubberband.cpp

```diff
@@ -207,6 +212,117 @@ void Rubberband::setWidthCurrentPoint( qreal width )
   emit widthCurrentPointChanged();
 }
 
+int Rubberband::traceTimeInterval() const
+{
+  return mTraceTimeInterval;
+}
+
+void Rubberband::setTraceTimeInterval( int traceTimeInterval )
+{
+  mTraceTimeInterval = traceTimeInterval;
+}
+
+int Rubberband::traceMinimumDistance() const
+{
+  return mTraceMinimumDistance;
+}
+
+void Rubberband::setTraceConjunction( bool traceConjunction )
+{
+  mTraceConjunction = traceConjunction;
+}
+
+bool Rubberband::traceConjunction() const
+{
+  return mTraceConjunction;
+}
+
+void Rubberband::setTraceMinimumDistance( int traceMinimumDistance )
+{
+  mTraceMinimumDistance = traceMinimumDistance;
+}
+
+void Rubberband::tracePosition()
+{
+  if ( std::isnan( model()->currentCoordinate().x() ) || std::isnan( model()->currentCoordinate().y() ) )
+  {
+    qDebug() << QString( "Coordinates not available " ) << " x:" << model()->currentCoordinate().x() << " y:" << model()->currentCoordinate().y();
+    return;
+  }
+  qDebug() << QString( "Collect " ) << model()->vectorLayer() << " x:" << model()->currentCoordinate().x() << " y:" << model()->currentCoordinate().y() << " z:" << model()->currentCoordinate().z();
+  model()->addVertex();
+  mTraceTimeIntervalFulfilled = false;
+  mTraceMinimumDistanceFulfilled = false;
+}
+
+void Rubberband::tracePositionReceived()
+{
+
+  QVector<QgsPointXY> points = mRubberbandModel->flatPointSequence( QgsProject::instance()->crs() );
+
+  auto pointIt = points.constEnd() - 1;
+
+  QVector<QgsPointXY> flatPoints;
+
+  flatPoints << *pointIt;
+  pointIt--;
+  flatPoints << *pointIt;
+
+  QgsDistanceArea distanceArea;
+  distanceArea.setEllipsoid( QgsProject::instance()->ellipsoid() );
+  distanceArea.setSourceCrs( QgsProject::instance()->crs(), QgsProject::instance()->transformContext() );
+
+  qDebug() << QString( "distance is: " ) << distanceArea.measureLine( flatPoints ) << QString( " and the minimum is " ) << mTraceMinimumDistance;
+
+  if ( distanceArea.measureLine( flatPoints ) > mTraceMinimumDistance )
+  {
+    mTraceMinimumDistanceFulfilled = true;
+    if ( !mTraceConjunction || mTraceTimeIntervalFulfilled )
+      tracePosition();
+  }
+}
+
+void Rubberband::traceTimeReceived()
+{
+  mTraceTimeIntervalFulfilled = true;
+  if ( !mTraceConjunction || mTraceMinimumDistanceFulfilled )
+    tracePosition();
+}
+
+void Rubberband::traceStart()
+{
+  if ( mTraceTimeInterval > 0 )
+  {
+    traceTimer = new QTimer( this );
```

Comment:
This leaks, no?
A `QTimer mTraceTimer;` in the header should be alright I guess?